[2022-09-27 19:09:18.704] [remoteagent] [info] Extension host agent started.
[2022-09-27 19:09:19.211] [remoteagent] [info] [127.0.0.1][d35e3921][ManagementConnection] New connection established.
[2022-09-27 19:09:19.288] [remoteagent] [info] [127.0.0.1][6c88bf1f][ExtensionHostConnection] New connection established.
[2022-09-27 19:09:19.306] [remoteagent] [info] [127.0.0.1][6c88bf1f][ExtensionHostConnection] <1661> Launched Extension Host Process.
[2022-09-27 19:39:06.161] [remoteagent] [warning] No ptyHost heartbeat after 6 seconds
[2022-09-27 19:39:20.471] [remoteagent] [warning] No ptyHost heartbeat after 6 seconds
[2022-09-27 19:39:33.947] [remoteagent] [error] ["CodeExpectedError: Could not find pty on pty host","    at c._throwIfNoPty (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:36316)","    at c.updateIcon (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:32124)","    at Object.call (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:16:8386)","    at u.onPromise (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5851)","    at u.onRawMessage (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5234)","    at /root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:4520","    at p.invoke (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:145)","    at D.deliver (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:2275)","    at T.fire (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:1853)","    at process.Q (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:9:20703)","    at process.emit (node:events:526:28)","    at emit (node:internal/child_process:938:14)","    at processTicksAndRejections (node:internal/process/task_queues:84:21)"]
[2022-09-27 19:39:33.950] [remoteagent] [error] ["CodeExpectedError: Could not find pty on pty host","    at c._throwIfNoPty (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:36316)","    at c.updateTitle (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:32064)","    at Object.call (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:16:8386)","    at u.onPromise (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5851)","    at u.onRawMessage (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5234)","    at /root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:4520","    at p.invoke (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:145)","    at D.deliver (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:2275)","    at T.fire (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:1853)","    at process.Q (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:9:20703)","    at process.emit (node:events:526:28)","    at emit (node:internal/child_process:938:14)","    at processTicksAndRejections (node:internal/process/task_queues:84:21)"]
[2022-09-27 19:39:33.991] [remoteagent] [error] ["CodeExpectedError: Could not find pty on pty host","    at c._throwIfNoPty (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:36316)","    at c.updateTitle (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:32064)","    at Object.call (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:16:8386)","    at u.onPromise (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5851)","    at u.onRawMessage (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5234)","    at /root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:4520","    at p.invoke (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:145)","    at D.deliver (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:2275)","    at T.fire (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:1853)","    at process.Q (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:9:20703)","    at process.emit (node:events:526:28)","    at emit (node:internal/child_process:938:14)","    at processTicksAndRejections (node:internal/process/task_queues:84:21)"]
[2022-09-27 19:39:37.022] [remoteagent] [info] [127.0.0.1][d35e3921][ManagementConnection] The client has reconnected.
[2022-09-27 19:39:37.298] [remoteagent] [error] ["CodeExpectedError: Could not find pty on pty host","    at c._throwIfNoPty (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:36316)","    at c.updateIcon (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:32124)","    at Object.call (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:16:8386)","    at u.onPromise (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5851)","    at u.onRawMessage (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5234)","    at /root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:4520","    at p.invoke (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:145)","    at D.deliver (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:2275)","    at T.fire (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:1853)","    at process.Q (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:9:20703)","    at process.emit (node:events:526:28)","    at emit (node:internal/child_process:938:14)","    at processTicksAndRejections (node:internal/process/task_queues:84:21)"]
[2022-09-27 19:39:37.298] [remoteagent] [error] ["CodeExpectedError: Could not find pty on pty host","    at c._throwIfNoPty (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:36316)","    at c.updateTitle (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:32064)","    at Object.call (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:16:8386)","    at u.onPromise (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5851)","    at u.onRawMessage (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5234)","    at /root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:4520","    at p.invoke (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:145)","    at D.deliver (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:2275)","    at T.fire (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:1853)","    at process.Q (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:9:20703)","    at process.emit (node:events:526:28)","    at emit (node:internal/child_process:938:14)","    at processTicksAndRejections (node:internal/process/task_queues:84:21)"]
[2022-09-27 19:39:38.113] [remoteagent] [error] ["CodeExpectedError: Could not find pty on pty host","    at c._throwIfNoPty (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:36316)","    at c.updateTitle (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:32064)","    at Object.call (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:16:8386)","    at u.onPromise (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5851)","    at u.onRawMessage (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5234)","    at /root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:4520","    at p.invoke (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:145)","    at D.deliver (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:2275)","    at T.fire (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:1853)","    at process.Q (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:9:20703)","    at process.emit (node:events:526:28)","    at emit (node:internal/child_process:938:14)","    at processTicksAndRejections (node:internal/process/task_queues:84:21)"]
[2022-09-27 19:39:39.215] [remoteagent] [error] ["CodeExpectedError: Could not find pty on pty host","    at c._throwIfNoPty (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:36316)","    at c.updateTitle (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:23:32064)","    at Object.call (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:16:8386)","    at u.onPromise (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5851)","    at u.onRawMessage (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:5234)","    at /root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:15:4520","    at p.invoke (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:145)","    at D.deliver (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:2275)","    at T.fire (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:11:1853)","    at process.Q (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/platform/terminal/node/ptyHostMain.js:9:20703)","    at process.emit (node:events:526:28)","    at emit (node:internal/child_process:938:14)","    at processTicksAndRejections (node:internal/process/task_queues:84:21)"]
[2022-09-27 19:39:40.734] [remoteagent] [info] [127.0.0.1][6c88bf1f][ExtensionHostConnection] The client has reconnected.
[2022-09-27 19:42:54.215] [remoteagent] [warning] No ptyHost heartbeat after 6 seconds
[2022-09-27 19:43:02.449] [remoteagent] [warning] No ptyHost heartbeat after 6 seconds
[2022-09-27 19:55:40.707] [remoteagent] [error] Error: connect ECONNREFUSED 127.0.0.1:3000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)
[2022-09-27 19:58:28.886] [remoteagent] [error] Error: connect ECONNREFUSED 127.0.0.1:3000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)
[2022-09-27 19:59:37.032] [remoteagent] [error] Error: connect ECONNREFUSED 127.0.0.1:3000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)
[2022-09-27 20:00:14.858] [remoteagent] [error] Error: connect ECONNREFUSED 127.0.0.1:3000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)
[2022-09-27 20:00:15.118] [remoteagent] [error] Error: connect ECONNREFUSED 127.0.0.1:3000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)
[2022-09-27 20:00:15.502] [remoteagent] [error] Error: connect ECONNREFUSED 127.0.0.1:3000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)
[2022-09-27 20:00:32.975] [remoteagent] [warning] No ptyHost heartbeat after 6 seconds
[2022-09-27 20:00:53.606] [remoteagent] [info] [127.0.0.1][d35e3921][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[2022-09-27 20:00:54.478] [remoteagent] [info] [127.0.0.1][6c88bf1f][ExtensionHostConnection] <1661> Extension Host Process exited with code: 0, signal: null.
[2022-09-27 20:00:54.478] [remoteagent] [info] Last EH closed, waiting before shutting down
[2022-09-27 20:00:58.452] [remoteagent] [error] Error: Unexpected SIGPIPE
    at process.<anonymous> (/root/.vscode-server/bin/e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb/out/vs/server/node/server.main.js:145:7922)
    at process.emit (node:events:526:28)
[2022-09-27 20:01:04.726] [remoteagent] [info] Got delay-shutdown request while in shutdown timeout, delaying
[2022-09-27 20:01:04.727] [remoteagent] [info] Cancelling previous shutdown timeout
[2022-09-27 20:01:04.908] [remoteagent] [info] [127.0.0.1][90433e51][ManagementConnection] New connection established.
[2022-09-27 20:01:04.982] [remoteagent] [info] [127.0.0.1][4ce003f6][ExtensionHostConnection] New connection established.
[2022-09-27 20:01:04.998] [remoteagent] [info] [127.0.0.1][4ce003f6][ExtensionHostConnection] <6140> Launched Extension Host Process.
